### PR TITLE
avoid throwing catching invalid group id exception

### PR DESCRIPTION
**Pull request by strokyl**
When starting a consumer without a group id, the code is going to throw and catch a lot of time the same exception. This is not really good for performance

### CODE REVIEW

File: src/main/scala/zio/kafka/consumer/internal/Runloop.scala [hunk 1]
```diff
@@ -218,9 +219,17 @@ private[consumer] final class Runloop(
...
-          concatenatedChunk.map(CommittableRecord(_, commit(_), Try(consumer.consumer.groupMetadata()).toOption))
-        )
+        fulfillAction = fulfillAction *> req.cont.succeed(concatenatedChunk.map { record =>
+          CommittableRecord(
+            record = record,
+            commitHandle = commit,
+            consumerGroupMetadata =
+              if (hasGroupId)
+                try Some(consumer.consumer.groupMetadata())
+                catch { case _: Throwable => None }
```
Comment (erikvanoosten) [inline src/main/scala/zio/kafka/consumer/internal/Runloop.scala:229]: Catching everything is not recommended
```suggestion
                catch { case NonFatal(_) => None }
```

Or simply: ```Try(consumer.consumer.groupMetadata()).toOption```

Comment (strokyl) [reply src/main/scala/zio/kafka/consumer/internal/Runloop.scala]: Well catch thanks!

Comment (strokyl) [reply src/main/scala/zio/kafka/consumer/internal/Runloop.scala]: `Try(..)` will allocate a try object to discard it right away, minimal optim but better not doing that somewhere we might pass a lot of time.

**Comment by strokyl** [comment]
Test failing:

>   - Consumer Streaming - partitions for topic doesn't fail if doesn't exist

Maybe it's a flaky?

**Comment by vigoo** [comment]
Isn't specifying group id mandatory?

**Comment by azhur** [comment]
@vigoo it's not. You can create a consumer without `group.id` when you don't need to spread the consumption to many consumers and don't want to store the offsets in Kafka.

**Comment by strokyl** [comment]
> Isn't specifying group id mandatory?

See:
https://github.com/zio/zio-kafka/issues/212
And:
https://github.com/zio/zio-kafka/pull/376
For more information about consuming without group id

**Comment by strokyl** [comment]
@erikvanoosten I reworked a bit the PR (see last commit) while rebasing

**Comment by strokyl** [comment]
Failing test:
```
 [info]       - it should not fail all the stream when error while fetching topic record stat
[info] Timeout of 30 s exceeded.
```

**Comment by erikvanoosten** [comment]
> @erikvanoosten Are you still interested in this contribution?

I have no say in that. I am not a maintainer of this project. 😀

**Review by vigoo — APPROVED** [review]

**Comment by vigoo** [comment]
Thanks for the explanation

**Pull request merged**